src/main/java/io/pravega/connectors/hadoop/PravegaOutputFormat.java

```diff
@@ -0,0 +1,176 @@
+                serializer = (PravegaEventRouter<V>) serializerClass.newInstance();
+            } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+                log.error("Exception when creating serializer: {}", e);
+                throw new IOException(
+                        "Unable to create the serializer for event router (" + eventRouterClassName + ")", e);
+            }
+        }
+        return serializer;
+    }
+
+    private EventStreamWriter<V> getPravegaWriter(
+            String scopeName,
+            String streamName,
+            URI controllerURI,
+            String serializerClassName) throws IOException {
+        ClientFactory clientFactory = (externalClientFactory != null) ? externalClientFactory : ClientFactory.withScope(scopeName, controllerURI);
+
+        Serializer serializer;
+
+        try {
+            Class<?> serializerClass = Class.forName(serializerClassName);
+            serializer = (Serializer<V>) serializerClass.newInstance();
+        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+            log.error("Exception when creating serializer: {}", e);
+            throw new IOException(
+                    "Unable to create the event serializer (" + serializerClassName + ")", e);
+        }
+
+        EventStreamWriter<V> writer = clientFactory.createEventWriter(streamName, serializer, EventWriterConfig.builder()
+                .transactionTimeoutTime(DEFAULT_TXN_TIMEOUT_MS)
+                .build());
+
+        return writer;
+    }
+
+    class PravegaOutputCommitter extends FileOutputCommitter {
+        PravegaOutputCommitter(Path outputPath, TaskAttemptContext context) throws IOException {
+            super(outputPath, context);
+        }
+
+        public void setupJob(JobContext context) throws IOException {
+            Configuration conf = context.getConfiguration();
+            final String scopeName = Optional.ofNullable(conf.get(PravegaConfig.OUTPUT_SCOPE_NAME)).orElseThrow(() ->
+                new IOException("The output scope name must be configured (" + PravegaConfig.OUTPUT_SCOPE_NAME + ")"));
+            final String streamName = Optional.ofNullable(conf.get(PravegaConfig.OUTPUT_STREAM_NAME)).orElseThrow(() ->
+                new IOException("The output stream name must be configured (" + PravegaConfig.OUTPUT_STREAM_NAME + ")"));
+            final URI controllerURI = Optional.ofNullable(conf.get(PravegaConfig.OUTPUT_URI_STRING)).map(URI::create).orElseThrow(() ->
+                new IOException("The Pravega controller URI must be configured (" + PravegaConfig.OUTPUT_URI_STRING + ")"));
+
+            int inputScaling = Integer.parseInt(conf.get(PravegaConfig.OUTPUT_SCALING, "0"));
+            final int scaling = inputScaling > 0 ? inputScaling : 1;
+
+            createStream(scopeName, streamName, controllerURI, scaling);
```

Comment:
Please ensure that `streamName` supports a fully-qualified stream name, with `scopeName` being an optional **default** scope to be used when the stream name is not qualified.

Reply:
(done)